generics/verification-steps.js

```diff
@@ -1,183 +1,71 @@
-
-        const optionsPromise = new Promise((res, rej) => {
-            select.findElements(by.tagName('option')).then((selections) => {
-                const numSelections = selections.length;
-                res(numSelections > 1);
-            });
-        });
-
-        expect(optionsPromise).to.eventually.be.ok.and.notify(next);
+        const selections = await select.findElements(by.tagName('option'));
+        const numSelections = selections.length;
+        if (numSelections <= 1) throw new Error(`${el} drop down is not showing options`);
     });
 
-    // Then(/^the "([^"]*)" text field is non-empty$/, (buttonText, next) => {
-    //   const buttonID = buttonText.replace(/ /g, '_').toUpperCase();
-    //   browser.wait(pageMap[global.pageID][buttonID].isPresent()).then(() => {
-    //     expect(pageMap[global.pageID][buttonID].getAttribute('value')).to.eventually.not.equal('').and.notify(next);
-    //   });
-    // });
-    //
-    // Then(/^the page (displays|does not display) "([^"]*)"$/, (torf, text, next) => {
-    //   const testEl = element(by.xpath(`//*[text()="${text}"]`));
-    //   testEl.isPresent().then((pres) => {
-    //     if (pres) {
-    //       expect(testEl.isDisplayed()).to.eventually.equal(torf === 'displays').and.notify(next);
-    //     } else {
-    //       expect(testEl.isPresent()).to.eventually.equal(torf === 'displays').and.notify(next);
-    //     }
-    //   });
-    // });
+    Then(/^the "([^"]*)" text field is non-empty$/, async (buttonText) => {
+        const buttonID = buttonText.replace(/ /g, '_').toUpperCase();
+        expect(pageMap[global.pageID][buttonID].getAttribute('value')).to.eventually.not.equal('');
+    });
 
-    Then(/^the "([^"]*)" element (is|is not) present$/, (elementText, yorn, next) => {
+    Then(/^the "([^"]*)" element (is|is not) present$/, async (elementText, yorn) => {
         const elementID = elementText.replace(/ /g, '_').toUpperCase();
+        const result = await driver.wait(pageMap[global.pageID][elementID].isDisplayed(), 5000);
 
-        driver.wait(pageMap[global.pageID][elementID].isDisplayed(), 5000)
-          .catch((err) => {
-              next(err);
-          })
-          .then((result) => {
-              if (yorn === 'is not') {
-                  if (result === true) next(new Error(`${elementID} Element is displayed`));
-                  else next();
-              }
-              else {
-                  if (result === true) next();
-                  else {
-                      next(new Error(`${elementID} Element is not displayed`));
-                  }
-              }
-          });
+        if (yorn === 'is not') {
+            if (result === true) throw new Error(`${elementID} Element is displayed`);
+        }
+        else {
+            if (result !== true) throw new Error(`${elementID} Element is not displayed`);
+        }
     });
 
-    // const validateImages = function (webTitle) {
```

Comment:
got rid of this, if we implement this in the near future, most likely we will do it in a different manner.